src/Raven.Server/Documents/Indexes/Static/IndexCompiler.cs

```diff
@@ -850,11 +858,12 @@ private static StatementSyntax HandleReduce(string reduce, FieldNamesValidator f
             var rewrittenExpression = (CSharpSyntaxNode)mapRewriter.Visit(expression);
 
             StatementSyntax optimized = null;
-
+            bool? foundSourceDocumentInProjection;
             try
             {
                 var visitor = new RavenLinqOptimizer(fieldValidator);
                 optimized = visitor.Visit(new RavenLinqPrettifier().Visit(rewrittenExpression)) as StatementSyntax;
+                foundSourceDocumentInProjection = visitor.FoundSourceDocumentInProjection;
```

Comment:
It looks like leftover from another work and it looks that it makes compilation errors

Reply:
done